prefect_azure/credentials.py

```diff
@@ -382,3 +382,24 @@ def example_get_workspace_flow():
         )
 
         return workspace
+
+
+class ContainerInstanceCredentials(Block):
+    """
+    Block used to manage Azure Container Instances authentication. Stores Azure Service
+    Principal authentication data.
+    """
+
+    _block_type_slug = "container-instance-credentials"
+    _block_type_name = "Azure Container Instance Credentials"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
+
+    client_id: str = Field(
+        default="...", title="Client ID", description="The service principal client ID."
+    )
+    tenant_id: str = Field(
+        default=..., title="Tenant ID", description="The service principal tenant ID."
+    )
+    client_secret: SecretStr = Field(
+        default=..., description="The service principal client secret."
+    )
```

Comment:
I'd like to see the client creation and authentication move from the container instance block to this block since I think it make more sense for this class to be responsible for authentication. A `get_client` method of some sort (similar to the other Azure credentials block) would be a good way to expose this functionality to consumers of this class.

Also, I think this class should be named `AzureContainerInstanceCredentials` to be consistent with the other Azure credentials blocks.

Reply:
I ended up adding two get client calls to the credentials block because we need two different clients to get everything done: A container instance client and an Azure resource client.